server/eb/app.py

```diff
@@ -40,14 +40,54 @@ def get_secret_key(region_name, secret_name, secret_key):
         if "SecretString" in get_secret_value_response:
             var = get_secret_value_response["SecretString"]
             secret = json.loads(var)
-            return secret.get(secret_key)
+            return secret
     except Exception:
         logging.critical("Caught exception during get_secret_key", exc_info=True)
         sys.exit(1)
 
     return None
 
 
+def handle_config_from_secret(app_config):
+    """Update configuration from the secret manager"""
+    secret_name = os.getenv("CXG_AWS_SECRET_NAME")
+    if not secret_name:
+        return
+
+    # need to find the secret manager region.
+    #  1. from CXG_AWS_SECRET_REGION_NAME
+    #  2. discover from dataroot location (if on s3)
+    #  3. discover from config file location (if on s3)
+    secret_region_name = os.getenv("CXG_AWS_SECRET_REGION_NAME")
+    if secret_region_name is None:
+        secret_region_name = discover_s3_region_name(app_config.multi_dataset__dataroot)
+        if not secret_region_name:
+            secret_region_name = discover_s3_region_name(config_file)
```

Comment:
could you set it as an env var so its available elsewhere in the app after this runs once?

Reply:
The secret_region_name is only used here to access the secrets manager